client/blocks/stats-navigation/index.js

```diff
@@ -17,24 +17,35 @@ import FollowersCount from 'blocks/followers-count';
 import { isSiteStore } from 'state/selectors';
 import { isJetpackSite } from 'state/sites/selectors';
 import { navItems, intervals as intervalConstants } from './constants';
+import config from 'config';
 
 class StatsNavigation extends Component {
 	static propTypes = {
 		interval: PropTypes.oneOf( intervalConstants.map( i => i.value ) ),
-		isJetpack: PropTypes.bool,
 		isStore: PropTypes.bool,
 		selectedItem: PropTypes.oneOf( Object.keys( navItems ) ).isRequired,
 		siteId: PropTypes.number,
 		slug: PropTypes.string,
 	};
 
+	isOverview = () => {
+		const { siteId } = this.props;
+		return 'undefined' === typeof siteId;
+	};
+
 	isValidItem = item => {
 		const { isStore, isJetpack } = this.props;
 		switch ( item ) {
-			case 'activity':
-				return isJetpack;
 			case 'store':
 				return isStore;
+			case 'activity':
+				if ( this.isOverview() ) {
```

Comment:
This method `isOverview` isn't used anywhere else so it could be deleted and this simplified to
```js
isValidItem = item => {
		const { isStore, isJetpack, siteId } = this.props;
		switch ( item ) {
			case 'store':
				return isStore;
			case 'activity':
				if ( 'undefined' === typeof siteId ) {
```

Reply:
agreed. it feels less magic by having a named method, though.